packages/frontend/core/src/components/workspace-selector/user-with-workspace-list/index.tsx

```diff
@@ -77,7 +76,7 @@ export const UserWithWorkspaceList = ({
   const onNewWorkspace = useCallback(() => {
     if (
       !isAuthenticated &&
-      !featureFlagService.flags.enable_local_workspace.value
+      !defaultServerService.server.config$.value.allowGuestDemoWorkspace
```

Comment:
It is necessary to consider compatibility with older versions; the older version of the server will not return this configuration, it will be `undefined`.

It should accurately determine that `allowGuestDemoWorkspace === false` to be considered as disabled